circus/controller.py

```diff
@@ -130,7 +130,12 @@ def stop(self):
         self.sys_hdl.stop()
 
     def handle_message(self, raw_msg):
-        cid, msg = raw_msg
+        try:
+            cid, msg = raw_msg
+        except:
```

Comment:
Could you catch only TypeError and ValueError ? As far as I can tell those are the only two exceptions that could happen here.